spring-web/src/main/java/org/springframework/http/codec/json/AbstractJackson2Decoder.java

```diff
@@ -126,7 +127,7 @@ public Flux<Object> decode(Publisher<DataBuffer> input, ResolvableType elementTy
 
 		ObjectMapper mapper = selectObjectMapper(elementType, mimeType);
 		if (mapper == null) {
-			throw new IllegalStateException("No ObjectMapper for " + elementType);
+			return Flux.error(new IllegalStateException("No ObjectMapper for " + elementType));
```

Comment:
This is arguably a "breaking change" to consumers of this method. It seems like the better option for error handling, but it may "break" anyone expecting to have to wrap the decode to handle it vs. handling it via Flux.